account_payment_method_or_mode_fs_storage/models/fs_storage.py

```diff
@@ -12,6 +12,9 @@ class FsStorage(models.Model):
     use_on_payment_method = fields.Boolean(
         help="Tells if storage can be set on payment methods",
     )
+    use_on_payment_mode = fields.Boolean(
```

Comment:
As discusses, IMO it's not a good idea to always adding a boolean on the storage itself to know where it should be available.
The idea is to add (into config) which storage are available for payment mode (and same for payment method).
The `use_on_payment_mode` shouldn't be an attribute of the storage.